src/codegen/llvm.zig

```diff
@@ -4035,28 +4035,29 @@ pub const Object = struct {
                     const struct_ty = try o.lowerType(ty);
                     if (struct_type.layout == .Packed) {
                         comptime assert(Type.packed_struct_layout_version == 2);
-                        var running_int = try o.builder.intConst(struct_ty, 0);
-                        var running_bits: u16 = 0;
-                        for (struct_type.field_types.get(ip), 0..) |field_ty, field_index| {
-                            if (!Type.fromInterned(field_ty).hasRuntimeBitsIgnoreComptime(mod)) continue;
-
-                            const non_int_val =
-                                try o.lowerValue((try val.fieldValue(mod, field_index)).toIntern());
-                            const ty_bit_size: u16 = @intCast(Type.fromInterned(field_ty).bitSize(mod));
-                            const small_int_ty = try o.builder.intType(ty_bit_size);
-                            const small_int_val = try o.builder.castConst(
-                                if (Type.fromInterned(field_ty).isPtrAtRuntime(mod)) .ptrtoint else .bitcast,
-                                non_int_val,
-                                small_int_ty,
-                            );
-                            const shift_rhs = try o.builder.intConst(struct_ty, running_bits);
-                            const extended_int_val =
-                                try o.builder.convConst(.unsigned, small_int_val, struct_ty);
-                            const shifted = try o.builder.binConst(.shl, extended_int_val, shift_rhs);
-                            running_int = try o.builder.binConst(.@"or", running_int, shifted);
-                            running_bits += ty_bit_size;
-                        }
-                        return running_int;
+                        const bytes = ty.abiSize(mod);
+
+                        var stack = std.heap.stackFallback(32, o.gpa);
+                        const allocator = stack.get();
+
+                        const limbs = try allocator.alloc(
+                            std.math.big.Limb,
+                            std.mem.alignForward(u64, bytes, @sizeOf(std.math.big.Limb)) / @sizeOf(std.math.big.Limb),
+                        );
+                        defer allocator.free(limbs);
+                        @memset(limbs, 0);
+
+                        val.writeToPackedMemory(
+                            ty,
+                            mod,
+                            std.mem.sliceAsBytes(limbs),
+                            0,
+                        ) catch unreachable;
+
+                        return o.builder.bigIntConst(struct_ty, .{
+                            .limbs = limbs,
+                            .positive = true,
+                        });
```

Comment:
This breaks when either the host or target is big endian, since `limbs` is a slice of native endian limbs in little endian order.